### PR TITLE
feat: expose all `SampleFile` options for `SampleReadme` in the constructor

### LINKED ISSUE #407: SampleReadme doesn't allow contents to be changed

### PR DESCRIPTION
**Pull request by pgollucci**
This will let things like `p6-projen-project-awesome-list` set their own `readme.md`
`filename` and `contents`.

Closes #407

### CODE REVIEW

File: test/samplereadme.test.ts [hunk 1]
```diff
@@ -36,9 +20,10 @@ describe('SampleReadProps', () => {
 
   test('customized w/ default SampleReadme', () => {
```
Comment (Chriscbr) [inline test/samplereadme.test.ts:21]: Can we omit this test? I'm not sure what different behavior it is testing from the test 'customized SampleReadme' below

File: test/samplereadme.test.ts [hunk 2]
```diff
@@ -1,28 +1,12 @@
-import * as path from 'path';
```
Comment (Chriscbr) [inline test/samplereadme.test.ts:1]: nit: Can we rename this file to `readme.test.ts` to make it clear it corresponds to the `readme.ts` source file? Or rename them `sample-readme(.test).ts`?

**Review by Chriscbr — CHANGES_REQUESTED** [review]

File: test/samplereadme.test.ts (hunk 1 above)
Comment (pgollucci) [reply test/samplereadme.test.ts:21]: It proves the default and back compat. That said, I can remove it.

File: test/samplereadme.test.ts (hunk 2 above)
Comment (pgollucci) [reply test/samplereadme.test.ts:1]: sure, I stuck with the original name of which I am not the creator

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**